### PR TITLE
OCPBUGS#8459: Add missing parameters for Azure

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.11+

Issue: https://issues.redhat.com/browse/OCPBUGS-8459

Link to docs preview: https://58870--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-customizations.html#installation-configuration-parameters-additional-azure_installing-azure-customizations

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -1727,6 +1727,50 @@ Additional Azure configuration parameters are described in the following table:
 |Defines the type of disk.
 |`premium_LRS` or `standardSSD_LRS`. The default is `premium_LRS`.
 
+|`platform.azure.defaultMachinePlatform.encryptionAtHost`
+|Enables host-level encryption for compute machines. You can enable this encryption alongside user-managed server-side encryption. This feature encrypts temporary, ephemeral, cached and un-managed disks on the VM host. This is not a prerequisite for user-managed server-side encryption.
+|`true` or `false`. The default is `false`.
+
+|`platform.azure.defaultMachinePlatform.osDisk`
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1734]: I think it doesn't need to mention this, since you already added its parameters under this object, `platform.azure.defaultMachinePlatform.osDisk.diskSizeGB` and `platform.azure.defaultMachinePlatform.osDisk.diskType`

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -1727,6 +1727,50 @@ Additional Azure configuration parameters are described in the following table:
...
+|
+|
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet`
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1738]: same as `platform.azure.defaultMachinePlatform.osDisk`, I think it doesn't need to mention it.

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -1727,6 +1727,50 @@ Additional Azure configuration parameters are described in the following table:
...
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.name`
+|The name of the disk encryption set that contains the encryption key from the installation prerequisites.
+|String, for example `production_disk_encryption_set`.
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.resourceGroup`
+|The name of the Azure resource group that contains the disk encryption set from the installation prerequisites. This resource group should be different from the resource group where you install the cluster to avoid deleting your Azure encryption key when the cluster is destroyed. This value is only necessary if you intend to install the cluster with user-managed disk encryption.
+|String, for example `production_encryption_resource_group`.
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.subscriptionId`
+|Optional. The ID of a disk encryption set in another Azure subscription. This secondary disk encryption set will be used to encrypt compute machines. By default, the installation program will use the disk encryption set from the Azure subscription ID that you provided to the installation program prompts.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1751]: Suggest to remove
`Optional`
`By default, the installation program will use the disk encryption set from the Azure subscription ID that you provided to the installation program prompts.`

as what you changes in https://github.com/openshift/openshift-docs/pull/58693

File: modules/installation-configuration-parameters.adoc [hunk 4]
```diff
@@ -1727,6 +1727,50 @@ Additional Azure configuration parameters are described in the following table:
...
+|`platform.azure.defaultMachinePlatform.type`
+|The Azure machine type for control plane and compute machines.
+|The Azure machine type.
+
+|`platform.azure.defaultMachinePlatform.vmNetworkingType`
+|The networking type for control plane and compute machines.
+|
+
+|`platform.azure.defaultMachinePlatform.zones`
+|For both compute machines and control plane machines, the list of availability zones that can be used. Examples: `cn-hangzhou-h`, `cn-hangzhou-j`
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1771]: How about rewording to `The availability zones where the installation program creates machines.`?

Comment (jinyunma) [reply modules/installation-configuration-parameters.adoc:1771]: I checked that `compute.platform.azure.zones` and `controlPlane.platform.azure.zones` are also missed, could you help to add them into the same table?

`compute.platform.azure.zones`
The availability zones where the installation program creates compute machines.

`controlPlane.platform.azure.zones` 
The availability zones where the installation program creates control plane machines.

Comment (xenolinux) [reply modules/installation-configuration-parameters.adoc]: @jinyunma Updated the PR. can you PTAL? Thanks for the review :)

File: modules/installation-configuration-parameters.adoc [hunk 5]
```diff
@@ -1727,6 +1731,46 @@ Additional Azure configuration parameters are described in the following table:
...
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.name`
+|The name of the disk encryption set that contains the encryption key from the installation prerequisites.
+|String, for example `production_disk_encryption_set`.
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.resourceGroup`
+|The name of the Azure resource group that contains the disk encryption set from the installation prerequisites. This resource group should be different from the resource group where you install the cluster to avoid deleting your Azure encryption key when the cluster is destroyed. This value is only necessary if you intend to install the cluster with user-managed disk encryption.
+|String, for example `production_encryption_resource_group`.
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.subscriptionId`
+|Optional. The ID of a disk encryption set in another Azure subscription. This secondary disk encryption set will be used to encrypt compute machines.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1751]: 1. Remove `Optional.`
2.  Reword `The ID of a disk encryption set in another Azure subscription.` -> `Defines the Azure subscription of the disk encryption set, where the disk encryption set resides.`, to keep consistent with change in https://github.com/openshift/openshift-docs/pull/58693/files#diff-14ebe576efb84b2f58b32a0de2996fced237204ae9b091e3cafbd40b65c1ef31R1073

**Comment by jinyunma** [comment]
@xenolinux thanks for addressing.

I also found below parameters are also missed, could you help to add them into the same table?
compute.platform.azure.type
controlPlane.platform.azure.type
platform.azure.defaultMachinePlatform.type

Here is an explanation from installer:
 ```
$ ./openshift-install explain installconfig.controlPlane.platform.azure.type
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <string>
  InstanceType defines the azure instance type. eg. Standard_DS_V2
```

File: modules/installation-configuration-parameters.adoc [hunk 6]
```diff
@@ -1719,6 +1719,14 @@ Additional Azure configuration parameters are described in the following table:
 |Defines the type of disk.
 |`standard_LRS`, `premium_LRS`, or `standardSSD_LRS`. The default is `premium_LRS`.
 
+|`compute.platform.azure.type`
+|Defines the azure instance type.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1723]: How about rewarding to `Defines the azure instance type for compute machines` ?

File: modules/installation-configuration-parameters.adoc [hunk 7]
```diff
@@ -1727,6 +1735,50 @@ Additional Azure configuration parameters are described in the following table:
 |Defines the type of disk.
 |`premium_LRS` or `standardSSD_LRS`. The default is `premium_LRS`.
 
+|`controlPlane.platform.azure.type`
+|Defines the azure instance type.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1739]: How about rewarding to Defines the azure instance type for control plane machines ?

File: modules/installation-configuration-parameters.adoc [hunk 8]
```diff
@@ -1727,6 +1735,50 @@ Additional Azure configuration parameters are described in the following table:
...
+|The Azure disk size for the VM.
+|Integer that represents the size of the disk in GB. The default is `128`.
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskType`
+|Defines the type of disk.
+|`standard_LRS`, `premium_LRS`, or `standardSSD_LRS`. The default is `premium_LRS`.
+
+|`platform.azure.defaultMachinePlatform.type`
+|The Azure machine type for control plane and compute machines.
+|The Azure machine type.
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1772]: How about changing `Azure machine type` to `Azure instance type` in line1771&line1772 to sync up with `compute.platform.azure.type` and `controlPlane.platform.azure.type`?

File: modules/installation-configuration-parameters.adoc [hunk 9]
```diff
@@ -1727,6 +1735,50 @@ Additional Azure configuration parameters are described in the following table:
...
+
+|`platform.azure.defaultMachinePlatform.vmNetworkingType`
```
Comment (jinyunma) [inline modules/installation-configuration-parameters.adoc:1774]: This parameter has already be tracked in https://github.com/openshift/openshift-docs/pull/58697, duplicated here.

**Comment by xenolinux** [comment]
@jinyunma Addressed the suggested change. Can you PTAL again?
Thank you

File: modules/installation-configuration-parameters.adoc [hunk 10]
```diff
@@ -1727,6 +1735,46 @@ Additional Azure configuration parameters are described in the following table:
...
+|The availability zones where the installation program creates control plane machines.
+|String list
+
+|`platform.azure.defaultMachinePlatform.encryptionAtHost`
+|Enables host-level encryption for compute machines. You can enable this encryption alongside user-managed server-side encryption. This feature encrypts temporary, ephemeral, cached and un-managed disks on the VM host. This is not a prerequisite for user-managed server-side encryption.
+|`true` or `false`. The default is `false`.
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.name`
+|The name of the disk encryption set that contains the encryption key from the installation prerequisites.
+|String, for example `production_disk_encryption_set`.
```
Comment (snarayan-redhat) [inline modules/installation-configuration-parameters.adoc:1752]:
```suggestion
|String, for example, `production_disk_encryption_set`.
```

File: modules/installation-configuration-parameters.adoc [hunk 11]
```diff
@@ -1727,6 +1735,46 @@ Additional Azure configuration parameters are described in the following table:
...
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.resourceGroup`
+|The name of the Azure resource group that contains the disk encryption set from the installation prerequisites. This resource group should be different from the resource group where you install the cluster to avoid deleting your Azure encryption key when the cluster is destroyed. This value is only necessary if you intend to install the cluster with user-managed disk encryption.
```
Comment (snarayan-redhat) [inline modules/installation-configuration-parameters.adoc:1755]: Apart from the suggestion below, I would still suggest to consider breaking this sentence. This is a complex one.

```suggestion
|The name of the Azure resource group that contains the disk encryption set from the installation prerequisites.  To avoid deleting your Azure encryption key when the cluster is destroyed, this resource group must be different from the resource group where you install the cluster. This value is only necessary if you intend to install the cluster with user-managed disk encryption.
```

File: modules/installation-configuration-parameters.adoc [hunk 12]
```diff
@@ -1727,6 +1735,46 @@ Additional Azure configuration parameters are described in the following table:
...
+|String, for example `production_encryption_resource_group`.
+
+|`platform.azure.defaultMachinePlatform.osDisk.diskEncryptionSet.subscriptionId`
+|Defines the Azure subscription of the disk encryption set where the disk encryption set resides. This secondary disk encryption set is used to encrypt compute machines.
```
Comment (snarayan-redhat) [inline modules/installation-configuration-parameters.adoc:1759]: Can you make the first sentence clearer? Does the suggestion make more sense?
```suggestion
|Defines the Azure subscription of the disk encryption set. This secondary disk encryption set is used to encrypt compute machines.
```

**Review by snarayan-redhat — COMMENTED** [review]
Looks great! Few suggestions.

File: modules/installation-configuration-parameters.adoc (hunk 12 above)
Comment (xenolinux) [reply modules/installation-configuration-parameters.adoc:1759]: Hmm, Actually IMO we should keep it as is to be consistent with this PR https://github.com/openshift/openshift-docs/pull/58693. 
Additional information: [1] https://github.com/openshift/openshift-docs/pull/58693#discussion_r1174500302
[2] https://github.com/openshift/openshift-docs/pull/58870#discussion_r1179850778

**Review by kelbrown20 — COMMENTED** [review]
Updates look great! 
Merging
